resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -52,29 +57,46 @@ protected void doResolve(final String inetHost, final Promise<InetAddress> promi
         // Note that InetSocketAddress.getHostName() will never incur a reverse lookup here,
         // because an unresolved address always has a host name.
-        resolveAll(inetHost).addListener(new FutureListener<List<InetAddress>>() {
-                    @Override
-                    public void operationComplete(Future<List<InetAddress>> future) throws Exception {
-                        if (future.isSuccess()) {
-                            List<InetAddress> inetAddresses = future.getNow();
-                            int numAddresses = inetAddresses.size();
-                            if (numAddresses == 0) {
-                                promise.setFailure(new UnknownHostException(inetHost));
-                            } else {
-                                // if there are multiple addresses: we shall pick one at random
-                                // this is to support the round robin distribution
-                                promise.setSuccess(inetAddresses.get(
-                                        numAddresses == 1 ? 0 : ThreadLocalRandom.current().nextInt(numAddresses)));
-                            }
-                        } else {
-                            promise.setFailure(future.cause());
-                        }
+        nameResolver.resolveAll(inetHost).addListener(new FutureListener<List<InetAddress>>() {
+            @Override
+            public void operationComplete(Future<List<InetAddress>> future) throws Exception {
+                if (future.isSuccess()) {
+                    List<InetAddress> inetAddresses = future.getNow();
+                    int numAddresses = inetAddresses.size();
+                    if (numAddresses >= 0) {
+                        // if there are multiple addresses: we shall pick one by one
+                        // to support the round robin distribution
+                        promise.setSuccess(inetAddresses.get(index++ % numAddresses));
+                    } else {
+                        promise.setFailure(new UnknownHostException(inetHost));
                     }
-                });
+                } else {
+                    promise.setFailure(future.cause());
+                }
+            }
+        });
     }
 
     @Override
-    protected void doResolveAll(String inetHost, Promise<List<InetAddress>> promise) throws Exception {
-        nameResolver.resolveAll(inetHost, promise);
+    protected void doResolveAll(String inetHost, final Promise<List<InetAddress>> promise) throws Exception {
+        nameResolver.resolveAll(inetHost).addListener(new FutureListener<List<InetAddress>>() {
+            @Override
+            public void operationComplete(Future<List<InetAddress>> future) throws Exception {
+                if (future.isSuccess()) {
+                    List<InetAddress> inetAddresses = future.getNow();
+                    if (inetAddresses.size() > 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1155)